Kubernetes/legos/k8s_get_expiring_certificates/k8s_get_expiring_certificates.py

```diff
@@ -41,32 +55,42 @@ def k8s_get_expiring_certificates(handle, namespace:str='', expiring_threshold:i
     """
     result = []
     all_namespaces = [namespace]
+
     cmd = "kubectl get ns  --no-headers -o custom-columns=':metadata.name'"
-    if namespace is None or len(namespace)==0:
-        kubernetes_namespaces = k8s_kubectl_command(handle=handle,kubectl_command=cmd )
-        replaced_str = kubernetes_namespaces.replace("\n"," ")
-        stripped_str = replaced_str.strip()
-        all_namespaces = stripped_str.split(" ")
+
+    if namespace is None or len(namespace) == 0:
+        response = handle.run_native_cmd(cmd)
+    if response is None:
+        print(
+            f"Error while executing command ({cmd}) (empty response)")
+
+    if response.stderr:
+        raise ApiException(
+            f"Error occurred while executing command {cmd} {response.stderr}")
+    kubernetes_namespaces = response.stdout
+    replaced_str = kubernetes_namespaces.replace("\n", " ")
+    stripped_str = replaced_str.strip()
+    all_namespaces = stripped_str.split(" ")
+
     coreApiClient = client.CoreV1Api(api_client=handle)
     for n in all_namespaces:
         coreApiClient.read_namespace_status(n, pretty=True)
-        secret = coreApiClient.list_namespaced_secret(n, watch=False, limit=200).items
-        expiration_threshold = timedelta(days=expiring_threshold)
-        # Check if the secret contains a certificate
-        if secret[0].type == "kubernetes.io/tls":
-            # Get the certificate data
-            cert_data = secret[0].data.get("tls.crt")
-            if cert_data:
-                # Decode the certificate data
-                cert_data_decoded = base64.b64decode(cert_data).decode("utf-8")
-                # Parse the certificate expiration date
-                cert_exp = None
-                try:
-                    cert_exp = datetime.strptime(cert_data_decoded.split("-----END CERTIFICATE-----")[0].split("Not After : ")[-1].strip(), "%b %d %H:%M:%S %Y %Z")
-                except ValueError:
-                    pass
-                if cert_exp and cert_exp < datetime.now() + expiration_threshold:
-                    result.append({"secret_name": secret[0].metadata.name, "namespace": n})
+        secrets = coreApiClient.list_namespaced_secret(n, watch=False, limit=200).items
```

Comment:
Assuming this will only pull 200 secrets at most.
ie the code will not function correctly for 201 secrets unless we add a iteration on `list_namespaced_secret()`?

Reply:
Yes it will process only the first 200 secrects. To fetch the rest of them we need a `continue_token`. The code will be in a while loop to fetch all values.

```
continue_token = None
    while True:
        secrets_response = coreApiClient.list_namespaced_secret(n, watch=False, limit=200, _continue=continue_token)
        secrets = secrets_response.items

        for secret in secrets:
            ...logic..
        if secrets_response.metadata and secrets_response.metadata._continue:
            continue_token = secrets_response.metadata._continue
        else:
            break
```